tests/userTest.js

```diff
@@ -638,8 +697,8 @@ describe('Main: Currently testing user management,', function () {
       return oc.users.setUserAttribute(config.nonExistentUser, 'email', 'asd@a.com').then(status => {
         expect(status).toBe(null)
       }).catch(error => {
-        expect(typeof (error)).toBe('object')
-        expect(error.ocs.meta.statuscode).toEqual('997')
+        expect(typeof (error)).toBe('string')
+        expect(error).toEqual('Unauthorized')
```

Comment:
why do we need this change?

Reply:
Yeah, It is a bit of surprising, as `setUserAttribute` is expected to return error object like in other tests but it actually returns string `Unauthorized` in case of non-existing-User

Reply:
https://github.com/owncloud/owncloud-sdk/blob/536e64836d4b51e1fa359943582b3d7a66a636f6/tests/userTest.js#L65-L74
The error is string `unauthorized` because the status code in interaction is `401` (returned by oc server when performing this request). But if the status code in interaction is set to `200`, the error returned is ocs object
Provider side testing will fail with statuscode `200` as it gets `401`

Reply:
`xyz` is non-existing user
```bash
curl -u admin:admin -X PUT http://localhost/core/ocs/v2.php/cloud/users/xyz -d key="email" -d value="example@mail.com" -v
```
```xml
* Server auth using Basic with user 'admin'
> PUT /core/ocs/v2.php/cloud/users/xyz HTTP/1.1
> Host: localhost
> Authorization: Basic YWRtaW46YWRtaW4=
> User-Agent: curl/7.68.0
> Accept: */*
> Content-Length: 32
> Content-Type: application/x-www-form-urlencoded
> 
< HTTP/1.1 401 Unauthorized
< Date: Wed, 17 Feb 2021 04:14:06 GMT
< Server: Apache/2.4.41 (Ubuntu)
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 0
< X-Robots-Tag: none
< X-Frame-Options: SAMEORIGIN
< X-Download-Options: noopen
< X-Permitted-Cross-Domain-Policies: none
< Set-Cookie: ocd3fn3x61nb=ebg5b0c8ve9j31n1qt6mo834g0; path=/core; HttpOnly
< Expires: Thu, 19 Nov 1981 08:52:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate
< Pragma: no-cache
< Set-Cookie: oc_sessionPassphrase=lU7XJYjUsaKSuVWHNz30NSViBI2OdHsCD1qOetX58XDk1KyLGoXbnBj%2Ft4fIR0CaSrf50CmF2VpIV5yMjWeSdHZbSwXIajKlxkCuB3IWp%2BXhRyZA51mQ0wQhhi8hXVnW; path=/core; HttpOnly; SameSite=strict
< Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *
< Set-Cookie: ocd3fn3x61nb=ln27kg3uh2emka932nj5fjjcq0; path=/core; HttpOnly
< Set-Cookie: cookie_test=test; expires=Wed, 17-Feb-2021 05:14:06 GMT; Max-Age=3600
< Set-Cookie: oc_username=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core; HttpOnly
< Set-Cookie: oc_token=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core; HttpOnly
< Set-Cookie: oc_remember_login=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core; HttpOnly
< Set-Cookie: oc_username=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core/; HttpOnly
< Set-Cookie: oc_token=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core/; HttpOnly
< Set-Cookie: oc_remember_login=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core/; HttpOnly
< Set-Cookie: ocd3fn3x61nb=bjbnlskdu92kg7ndvurts6gfcp; path=/core; HttpOnly
< Set-Cookie: ocd3fn3x61nb=7dvt62k1j6f73mpccdo4levnq8; path=/core; HttpOnly
* Authentication problem. Ignoring this.
< WWW-Authenticate: Basic realm="Authorisation Required"
< Content-Length: 132
< Content-Type: text/xml; charset=UTF-8
< 
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>997</statuscode>
  <message/>
 </meta>
 <data/>
</ocs>
```

Reply:
provider's `willRespondWith` doesn't respond provided `body` when statucode is `401`

Reply:
the status code should be `200` for v1 API and `401` for v2 requests
the interaction is set to use `v1` https://github.com/owncloud/owncloud-sdk/pull/714/files#diff-b799f43eb8d74a22d0a06495f79b2b1ec471eaa171dc202b5d564fb9ebca4064R80
your curl request uses `v2`

Reply:
I also thought `v2` may be the problem, but same result with `v1` also

Reply:
```
curl -u admin:admin -X PUT http://localhost/core/ocs/v1.php/cloud/users/xyz -d key="email" -d value="example@mail.com" -v

*   Trying 127.0.0.1:80...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 80 (#0)
* Server auth using Basic with user 'admin'
> PUT /core/ocs/v1.php/cloud/users/xyz HTTP/1.1
> Host: localhost
> Authorization: Basic YWRtaW46YWRtaW4=
> User-Agent: curl/7.68.0
> Accept: */*
> Content-Length: 32
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 32 out of 32 bytes
* Mark bundle as not supporting multiuse
< HTTP/1.1 401 Unauthorized
< Date: Wed, 17 Feb 2021 06:02:20 GMT
< Server: Apache/2.4.41 (Ubuntu)
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 0
< X-Robots-Tag: none
< X-Frame-Options: SAMEORIGIN
< X-Download-Options: noopen
< X-Permitted-Cross-Domain-Policies: none
< Set-Cookie: ocd3fn3x61nb=1hnt3f4v6derp3tma8ajpo6247; path=/core; HttpOnly
< Expires: Thu, 19 Nov 1981 08:52:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate
< Pragma: no-cache
< Set-Cookie: oc_sessionPassphrase=9dGRD6bVRjpAenJe6RzikKSETdMNTxt9puIp%2FqmDnY12b4XnGLGiFWk%2F8KVwMSq6WOsoYw9QgGf0Ums9c7hKoxnAANWsxvwiqGFjFk4OpLOQN%2FQW0UuX%2Fm6GnX64rS27; path=/core; HttpOnly; SameSite=strict
< Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *
< Set-Cookie: ocd3fn3x61nb=8t462ol86ohc3smf7k49gksrnc; path=/core; HttpOnly
< Set-Cookie: cookie_test=test; expires=Wed, 17-Feb-2021 07:02:20 GMT; Max-Age=3600
< Set-Cookie: oc_username=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core; HttpOnly
< Set-Cookie: oc_token=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core; HttpOnly
< Set-Cookie: oc_remember_login=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core; HttpOnly
< Set-Cookie: oc_username=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core/; HttpOnly
< Set-Cookie: oc_token=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core/; HttpOnly
< Set-Cookie: oc_remember_login=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0; path=/core/; HttpOnly
< Set-Cookie: ocd3fn3x61nb=pqgleurba0bt1tkedadqttuba6; path=/core; HttpOnly
< Set-Cookie: ocd3fn3x61nb=hicoltb4584aokdr581ofhc2ni; path=/core; HttpOnly
* Authentication problem. Ignoring this.
< WWW-Authenticate: Basic realm="Authorisation Required"
< Content-Length: 132
< Content-Type: text/xml; charset=UTF-8
< 
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>997</statuscode>
  <message/>
 </meta>
 <data/>
</ocs>
* Connection #0 to host localhost left intact
```